pkg/helpers/queuekey.go

```diff
@@ -20,15 +20,30 @@ const (
 	BootstrapHubKubeConfig = "bootstrap-hub-kubeconfig"
 	// HubKubeConfig is the secret name of kubeconfig secret to connect to hub with mtls
 	HubKubeConfig = "hub-kubeconfig-secret"
-	// ClusterManagerNamespace is the namespace to deploy cluster manager components
-	ClusterManagerNamespace = "open-cluster-management-hub"
+	// ClusterManagerNamespaceSuffix is the suffix of the namespace to deploy cluster manager components
+	ClusterManagerNamespaceSuffix = "open-cluster-management-hub"
 
 	RegistrationWebhookSecret  = "registration-webhook-serving-cert"
 	RegistrationWebhookService = "cluster-manager-registration-webhook"
 	WorkWebhookSecret          = "work-webhook-serving-cert"
 	WorkWebhookService         = "cluster-manager-work-webhook"
 )
 
+func ClusterManagerNamespaceWithDefaultMode(clustermanagername string) string {
+	return "open-cluster-management-hub"
+}
+
+func ClusterManagerNamespace(clustermanagername string, mode operatorapiv1.InstallMode) string {
+	if mode == operatorapiv1.InstallModeDetached {
+		return clustermanagername + "-" + ClusterManagerNamespaceSuffix
```

Comment:
this could be pretty long. We do not really need this as suffix, do we?

Reply:
Yes, we don't really needs this.

Reply:
Done. Use cluster-manager name as namespace only.